config/routes.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+Rails.application.routes.draw do
+  root 'users#index'
+  resources :users, only: %i[index show]
+  resources :user_posts, only: %i[index show]
```

Comment:
- [ ] [Optional] Since posts belongs to users, you should have nested resources. This will be:
```ruby
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
end
```

Reply:
Thank you, I'll work on this.